### PR TITLE
Replace Transaction::fee with ::fee_in and ::all_fees 

### PR DESCRIPTION
**Pull request by stevenroose**
@RCasatta pointed out that we don't check the asset type on fee outputs. In theory they can be any asset.

### CODE REVIEW

**Comment by stevenroose** [comment]
As a follow-up I'd like to change the internal type for `confidential::Asset` from `sha256d::Hash` which is just wrong to `AssetId`.

File: src/transaction.rs [hunk 1]
```diff
@@ -613,10 +615,25 @@ impl Transaction {
         bitcoin::Wtxid::from_engine(enc)
     }
 
-    /// Get the total transaction fee.
-    pub fn fee(&self) -> u64 {
+    /// Get the total transaction fee in the given asset.
+    pub fn fee_in(&self, asset: AssetId) -> u64 {
```
Comment (RCasatta) [inline src/transaction.rs:619]: maybe simpler calling all_fees() ?

File: src/transaction.rs [hunk 2]
```diff
@@ -613,10 +615,25 @@ impl Transaction {
...
+            .filter(|o| o.is_fee() && o.asset == asset)
+            .filter_map(|o| o.value.explicit())
+            .sum()
+    }
+
+    /// Get all fees in all assets.
+    pub fn all_fees(&self) -> HashMap<AssetId, u64> {
+        let mut fees = HashMap::new();
+        for out in self.output.iter().filter(|o| o.is_fee()) {
+            let asset = AssetId::from_inner(sha256::Midstate::from_inner(out.asset.explicit().unwrap().into_inner()));
```
Comment (RCasatta) [inline src/transaction.rs:632]: maybe a comment why is safe to unwrap() ?

File: src/transaction.rs (hunk 1 above)
Comment (stevenroose) [reply src/transaction.rs:619]: I'd like to avoid building the hashmap etc.

File: src/transaction.rs (hunk 2 above)
Comment (stevenroose) [reply src/transaction.rs:632]: good point, I have that comment in fee_in, but not here, will add.

**Comment by RCasatta** [comment]
ack 16025626253f5c3414ab4d584fa6d0e041b12b7d

**Comment by stevenroose** [comment]
Travis is acting out, this passed CI here: https://travis-ci.org/github/ElementsProject/rust-elements/builds/670508921.

**Pull request merged**